src/main/java/seedu/address/logic/parser/AddFloatingTaskParser.java

```diff
@@ -67,8 +69,13 @@ private static ParseResult parseArguments(String str) throws IllegalValueExcepti
         if (args.isEmpty()) {
             return result;
         }
-        if (Priority.isValidPriority(args.get(0))) {
-            result.priority = args.remove(0);
+        if (isPriorityFormat(args.get(0))) {
+            Matcher matcher = PRIORITY_PATTERN.matcher(args.remove(0));
+            matcher.find();
+            result.priority = matcher.group("priority");
+            if (!Priority.isValidPriority(result.priority)) {
+                throw new IllegalValueException("priority can only range from 0 to 5");
```

Comment:
Won't the `Priority` constructor already catch this? Why duplicate exception messages?